### PR TITLE
fix(ci): add markdownlint custom rule for plain code fences

### PR DESCRIPTION
**Pull request by OnkarRuikar**
Makes https://github.com/mdn/content/pull/35140 permanent.

In yari we have:
```js
// Over the years we have accumulated some weird <pre> tags whose
// brush is more or less "junk".
// TODO: Perhaps, if you have a doc with <pre> tags that matches
// this, it should become a flaw.
const IGNORE = new Set(["none", "text", "plain", "unix"]);
```
We need to keep only `plain` and ban rest.

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/json/stringify/index.md [hunk 1]
```diff
@@ -270,8 +270,8 @@ Using a tab character mimics standard pretty-print appearance:
 console.log(JSON.stringify({ uno: 1, dos: 2 }, null, "\t"));
 /*
 {
-	"uno": 1,
-	"dos": 2
+  "uno": 1,
+  "dos": 2
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:274]: Could you revert this change? I'm not sure why this often gets randomly fixed but not always.

Comment (OnkarRuikar) [inline files/en-us/web/javascript/reference/global_objects/json/stringify/index.md:274]:
```suggestion
	"uno": 1,
	"dos": 2
```

Comment (OnkarRuikar) [reply files/en-us/web/javascript/reference/global_objects/json/stringify/index.md]: Done. Some editors convert tabs to space automatically on save. I have no idea which one did this.

**Comment by Josh-Cena** [comment]
Have you tested locally to see if it works? I'm not sure if `"searchScope": "text"` searches just plain text (i.e. not Markdown syntax) or if it searches everything.

**Comment by OnkarRuikar** [comment]
> Have you tested locally to see if it works? I'm not sure if `"searchScope": "text"` searches just plain text (i.e. not Markdown syntax) or if it searches everything.

I added the rule to the config file and ran the markdown linter. The changes suggested by the linter were pulled in https://github.com/mdn/content/pull/35140 .

The search scope `text` searches everything except inline and block code fences. Do you have any case that is failing?

File: .markdownlint.jsonc [hunk 2]
```diff
@@ -212,6 +212,13 @@
         "searchPattern": "/(\\[[^\\]]*?\\]\\(([^\\)\\n]|\\([^\\)\\n]\\)|\\s[\"'])+?)(\\n|\\s|[,:][\\s\\n])/gm",
         "searchScope": "text",
       },
+      {
+        "name": "text-code-fence",
+        "message": "Use 'plain' tag",
+        "searchPattern": "/(^ *```)(text|none|unix)/gm",
```
Comment (Josh-Cena) [inline .markdownlint.jsonc:218]:
```suggestion
        "searchPattern": "/(^ *`{3,})(text|none|unix)/gm",
```

A code fence can have more than 3 backticks.

**Comment by Josh-Cena** [comment]
No; I was just wondering if it works. If that PR was generated using this rule, then this certainly looks good to me.

**Review by Josh-Cena — APPROVED** [review]

**Comment by Josh-Cena** [comment]
Works great, thanks!
